docs/reference/modules/cluster/shards_allocation.asciidoc

```diff
@@ -97,9 +97,22 @@ Specify when shard rebalancing is allowed:
 [[shards-rebalancing-heuristics]]
 ==== Shard balancing heuristics settings
 
-The following settings are used together to determine where to place each
-shard.  The cluster is balanced when no allowed rebalancing operation can bring the weight
-of any node closer to the weight of any other node by more than the `balance.threshold`.
+A cluster is _balanced_ when it has an equal number of shards on each node
+without having a concentration of shards from any index on any node. {es} runs
+an automatic process called _rebalancing_ which moves shards between the nodes
+in your cluster in order to improve its balance. Rebalancing obeys all other
+shard allocation rules such as <<cluster-shard-allocation-filtering,allocation
+filtering>> and <<forced-awareness,forced awareness>> which may prevent it from
+completely balancing the cluster. In that case, rebalancing strives to acheve
+the most balanced cluster possible within the rules you have configured.
```

Comment:
Feels like we should mention data tiers here. In many cases, I imagine the tiers, rather than the cluster, will be balanced.